SimpleAT.c

```diff
@@ -9,6 +10,24 @@ static uint8_t __sizeOfEngine;
 #define LOG(...)
 #endif
 
+#define STATE_WAIT_A 0
+#define STATE_WAIT_T 1
+#define STATE_WAIT_PLUS 2
+#define STATE_COMMAND 3
+#define STATE_CHECK_COMMAND 4
+#define STATE_PARAMS 5
+#define STATE_STRING_PARAM 6
+#define STATE_ERROR 255
```

Comment:
I think enum is better than define in this case, but it is already a lot better than original ;)

Reply:
Since this is only being used internally and is not passed as a parameter or function return, I have chosen to use defines to be as light as possible. Although enum is converted to number right?